xoa/data_source_host.go

```diff
@@ -0,0 +1,36 @@
+package xoa
+
+import (
+	"github.com/ddelnano/terraform-provider-xenorchestra/client"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceXoaHost() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceHostRead,
+		Schema: map[string]*schema.Schema{
+			"name_label": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func dataSourceHostRead(d *schema.ResourceData, m interface{}) error {
+	c := m.(*client.Client)
+	nameLabel := d.Get("name_label").(string)
+	hosts, err := c.GetHostByName(nameLabel)
+	if err != nil {
+		return err
+	}
+
+	if _, ok := err.(client.NotFound); ok {
+		d.SetId("")
+		return nil
+	}
+
+	d.SetId(hosts[0].Id)
```

Comment:
We need to handle the condition where there are more than 1 host returned. While I think it would be odd for multiple hosts to have the same name label, it is technically possible for name labels to not be unique.

There is other code that handles this correctly (see [this](https://github.com/terra-farm/terraform-provider-xenorchestra/blob/d903d73146394d0cd3651707ba7ab40157c9a840/xoa/data_source_cloud_config.go#L44-L47) as an example and the [tests](https://github.com/terra-farm/terraform-provider-xenorchestra/blob/d903d73146394d0cd3651707ba7ab40157c9a840/xoa/data_source_cloud_config_test.go#L62)).

Reply:
I added the `len(hosts) == 1` to match the cloud config datasource. 

Testing multiple host names may not be that easy. I could not find a way to "create" two xcp hosts with the same name. It might be a better solution to implement smth. like a XOA stub ? Maybe you can come up with a better solution for testing this.

Reply:
If it was convenient to test this condition, I would say it would be worth it, however, I think we can skip testing this situation for now. If there is ever a regression related to it than we can brainstorm the best way to test it.